crates/hyperswitch_connectors/src/connectors/deutschebank/transformers.rs

```diff
@@ -651,7 +934,12 @@ impl TryFrom<&DeutschebankRouterData<&PaymentsCaptureRouterData>> for Deutscheba
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             changed_amount: item.amount,
-            kind: DeutschebankTransactionKind::Directdebit,
+            kind: match item.router_data.is_three_ds()
+                && matches!(item.router_data.payment_method, PaymentMethod::Card)
+            {
+                true => DeutschebankTransactionKind::Creditcard3ds20,
+                false => DeutschebankTransactionKind::Directdebit,
+            },
```

Comment:
Add explicit matching where
 if sepa then value should be Directdebit
if cards + 3ds the value should be  Creditcard3ds20
else not supported error